UPGRADE.md

```diff
@@ -1,5 +1,42 @@
 # Upgrade to 3.0
 
+## BC BREAK: New argument to `NamingStrategy::joinColumnName()`
+
+### Before
+
+```php
+<?php
+class MyStrategy implements NamingStrategy
+{
+    /**
+     * @param string $propertyName A property name.
+     */
+    public function joinColumnName($propertyName): string
+    {
+        // …
+    }
+}
+```
+
+### After
+
+The `class-string` type for `$className` can be inherited from the signature of
+the interface.
+
+```php
+<?php
+class MyStrategy implements NamingStrategy
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function joinColumnName(string $propertyName, string $className = null): string
```

Comment:
`$propertyName` does not (yet) have a parameter type on the `NamingStrategy` interface. Adding the type here would cause a fatal error. I suggest to add the type to the interface and the implementations.

Moreover, the second argument is not supposed to be optional. Why make it optional in this example?

Reply:
> Moreover, the second argument is not supposed to be optional. Why make it optional in this example?

I copy/pasted what I did for the deprecation and forgot to remove that.

Reply:
Regarding the type declarations, I think I will address this for the whole classes and interface separately, after https://github.com/doctrine/orm/pull/9756 is merged up. Let's pause this for now.

Reply:
It's done now :)

Reply:
```suggestion
    public function joinColumnName(string $propertyName, string $className): string
```